### PR TITLE
Update doc - Making a Shelley blockchain from scratch

### PR DESCRIPTION
**Pull request by ArturWieczorek**

### CODE REVIEW

**Comment by ArturWieczorek** [comment]
This part caught my interest:
```
For the configuration file, we can start with the default

$ cp cardano-node/configuration/defaults/byron-mainnet/configuration.yaml \
     example/

and make a couple tweaks

$ sed -i 's/Protocol: RealPBFT/Protocol: TPraos/' example/configuration.yaml
$ sed -i 's/minSeverity: Info/minSeverity: Debug/' example/configuration.yaml

We will share this `configuration.yaml` file between both nodes we run.
```

as after making those changes and starting nodes socket files were not created.
First thing is that I didn't see any information in the logs about issues with socket not being created - not sure if this is OK ? 
```
$ cardano-node run     --config example/configuration.yaml     --topology example/node2/topology.json     --database-path example/node2/db     --socket-path example/node2/node.sock     --shelley-kes-key example/node2/kes.skey     --shelley-vrf-key example/node2/vrf.skey     --shelley-operational-certificate example/node2/cert     --port 3002
[artur-de:cardano.node:Debug:5] [2020-05-28 10:09:23.17 UTC] tracing verbosity = normal
[artur-de:cardano.node:Debug:5] [2020-05-28 10:09:23.18 UTC] System started at SystemStart {getSystemStart = 2020-05-28 10:00:23.004438246 UTC}
[artur-de:cardano.node:Debug:5] [2020-05-28 10:09:23.18 UTC] 
**************************************
Host node address: Just (NodeAddress {naHostAddress = Nothing, naPort = 3002})
My DNS producers are []
My IP producers are [NodeAddress {naHostAddress = Just 127.0.0.1, naPort = 3001}]
**************************************

[artur-de:cardano.node.release:Notice:5] [2020-05-28 10:09:23.18 UTC] TPraos
[artur-de:cardano.node.release:Notice:5] [2020-05-28 10:09:23.18 UTC] NetworkMagic 42
[artur-de:cardano.node.version:Notice:5] [2020-05-28 10:09:23.18 UTC] 1.12.0
[artur-de:cardano.node.commit:Notice:5] [2020-05-28 10:09:23.18 UTC] f6ef186d1f7280585a245c8565d92117133cf359
[artur-de:cardano.node.Forge:Info:37] [2020-05-28 10:09:24.67 UTC] Testing for leadership at slot 541
[artur-de:cardano.node.ForgeTime:Info:37] [2020-05-28 10:09:24.67 UTC] fromList []
[artur-de:cardano.node.Forge:Info:37] [2020-05-28 10:09:24.67 UTC] Not leading slot 541
[artur-de:cardano.node.ForgeTime:Info:37] [2020-05-28 10:09:24.67 UTC] fromList []
[artur-de:cardano.node.IpSubscription:Debug:50] [2020-05-28 10:09:24.67 UTC] [String "Starting Subscription Worker, valency 1",String "[127.0.0.1:3001]",String "WithIPList SubscriptionTrace",String "LocalAddresses {laIpv4 = Just 0.0.0.0:0, laIpv6 = Just [::]:0, laUnix = Nothing}"]
[artur-de:cardano.node.IpSubscription:Info:50] [2020-05-28 10:09:24.67 UTC] [String "Trying to connect to 127.0.0.1:3001",String "[127.0.0.1:3001]",String "WithIPList SubscriptionTrace",String "LocalAddresses {laIpv4 = Just 0.0.0.0:0, laIpv6 = Just [::]:0, laUnix = Nothing}"]
[artur-de:cardano.node.IpSubscription:Info:52] [2020-05-28 10:09:24.67 UTC] [String "Connection Attempt Start, destination 127.0.0.1:3001",String "[127.0.0.1:3001]",String "WithIPList SubscriptionTrace",String "LocalAddresses {laIpv4 = Just 0.0.0.0:0, laIpv6 = Just [::]:0, laUnix = Nothing}"]
[artur-de:cardano.node.IpSubscription:Debug:52] [2020-05-28 10:09:24.67 UTC] [String "Allocate socket to 127.0.0.1:3001",String "[127.0.0.1:3001]",String "WithIPList SubscriptionTrace",String "LocalAddresses {laIpv4 = Just 0.0.0.0:0, laIpv6 = Just [::]:0, laUnix = Nothing}"]
[artur-de:cardano.node.IpSubscription:Notice:50] [2020-05-28 10:09:24.67 UTC] [String "Waiting 0.025s before attempting a new connection",String "[127.0.0.1:3001]",String "WithIPList SubscriptionTrace",String "LocalAddresses {laIpv4 = Just 0.0.0.0:0, laIpv6 = Just [::]:0, laUnix = Nothing}"]
[artur-de:cardano.node.IpSubscription:Notice:52] [2020-05-28 10:09:24.67 UTC] [String "Connection Attempt End, destination 127.0.0.1:3001 outcome: ConnectSuccessLast",String "[127.0.0.1:3001]",String "WithIPList SubscriptionTrace",String "LocalAddresses {laIpv4 = Just 0.0.0.0:0, laIpv6 = Just [::]:0, laUnix = Nothing}"]
[artur-de:cardano.node.IpSubscription:Debug:50] [2020-05-28 10:09:24.70 UTC] [String "Required subscriptions started",String "[127.0.0.1:3001]",String "WithIPList SubscriptionTrace",String "LocalAddresses {laIpv4 = Just 0.0.0.0:0, laIpv6 = Just [::]:0, laUnix = Nothing}"]
[artur-de:cardano.node.Forge:Info:37] [2020-05-28 10:09:25.00 UTC] Testing for leadership at slot 542
[artur-de:cardano.node.ForgeTime:Info:37] [2020-05-28 10:09:25.00 UTC] fromList []
[artur-de:cardano.node.Forge:Info:37] [2020-05-28 10:09:25.00 UTC] Not leading slot 542
[artur-de:cardano.node.ForgeTime:Info:37] [2020-05-28 10:09:25.00 UTC] fromList []
[artur-de:cardano.node.Forge:Info:37] [2020-05-28 10:09:26.00 UTC] Testing for leadership at slot 543
[artur-de:cardano.node.ForgeTime:Info:37] [2020-05-28 10:09:26.00 UTC] fromList []
[artur-de:cardano.node.Forge:Info:37] [2020-05-28 10:09:26.00 UTC] Not leading slot 543
[artur-de:cardano.node.ForgeTime:Info:37] [2020-05-28 10:09:26.00 UTC] fromList []
[artur-de:cardano.node.Forge:Info:37] [2020-05-28 10:09:27.00 UTC] Testing for leaders
```
and content of dir showing that there is no socket :

```
example/node2$ ls -al
total 36
drwxr-xr-x 3 artur artur 4096 May 28 12:09 .
drwxr-xr-x 7 artur artur 4096 May 28 12:09 ..
-rw-r--r-- 1 artur artur  337 May 28 12:00 cert
drwxr-xr-x 5 artur artur 4096 May 28 12:09 db
-rw-r--r-- 1 artur artur 1517 May 28 12:00 kes.skey
-rw-r--r-- 1 artur artur  142 May 28 12:00 kes.vkey
-rw-r--r-- 1 artur artur  103 May 28 11:40 topology.json
-rw-r--r-- 1 artur artur   96 May 28 12:00 vrf.skey
-rw-r--r-- 1 artur artur  134 May 28 12:00 vrf.vkey
```

After commenting `SocketPath` line in the *configuration.yaml* that those nodes are using:

```
##### Locations #####

GenesisFile: genesis.json
#SocketPath: db/node.socket
```
sockets were created so this raise second question - should that file take precedence over command line arguments, is this correct behavior ?

File: doc/shelley-genesis.md [hunk 1]
```diff
@@ -1084,14 +1092,14 @@ the `genesis.json`.
 There is a command to query the UTxO and return all the UTxOs at a given
 address.
 ```
-$ cardano-cli shelley query filtered-utxo
-Usage: cardano-cli shelley query filtered-utxo --address ADDRESS
-                                               (--mainnet | --testnet-magic INT)
-                                               [--out-file FILE]
-  Get the node's current UTxO filtered by address
+$ cardano-cli shelley query utxo
+Usage: cardano-cli shelley query utxo [--address ADDRESS]
```
Comment (ArturWieczorek) [inline doc/shelley-genesis.md:1096]: In this line address in now in square brackets which suggests that it is optional param [--address ADDRESS] So this should probably be changed.

**Comment by dcoutts** [comment]
> as after making those changes and starting nodes socket files were not created.

This was fixed in 1.13.0.

File: doc/shelley-genesis.md (hunk 1 above)
Comment (dcoutts) [reply doc/shelley-genesis.md:1096]: What should be changed exactly? Yes, the address flag is optional. You can indeed ask for the whole UTxO.

Comment (ArturWieczorek) [reply doc/shelley-genesis.md:1096]: My bad, I though that some address should be specified, it's an old PR and I forgot about it. I will close it.

**Pull request closed without merging**